src/components/GitHubSync.tsx

```diff
@@ -0,0 +1,992 @@
+  if (!enabled) return;
+  const gitHubNodeResult = window.roamAlphaAPI.data.fast.q(`[:find
+    (pull ?node [:block/string])
+    :where
+      [?roamjsgithub-sync :node/title "roam/js/github-sync"]
+      [?node :block/page ?roamjsgithub-sync]
+      [?p :block/children ?node]
+    (or     [?p :block/string ?p-String]
+      [?p :node/title ?p-String])
+    [(clojure.string/includes? ?p-String "Node Select")]
+    ]`) as [PullBlock][];
+  const nodeText = gitHubNodeResult[0]?.[0]?.[":block/string"] || "";
+  if (!nodeText) return;
+
+  const discourseNodes = getDiscourseNodes();
+  const selectedNode = discourseNodes.find((node) => node.text === nodeText);
+  const isPageTypeOfNode = matchDiscourseNode({
+    format: selectedNode?.format || "",
+    specification: selectedNode?.specification || [],
+    text: selectedNode?.text || "",
+    title: pageTitle,
+  });
+  return isPageTypeOfNode;
+};
+
+export const renderGitHubSyncPage = async ({
+  h1,
+  pageUid,
+  onloadArgs,
+}: {
+  h1: HTMLHeadingElement;
+  pageUid: string;
+  onloadArgs: OnloadArgs;
+}) => {
+  const extensionAPI = onloadArgs.extensionAPI;
+
+  const commentsContainerUid = await getRoamCommentsContainerUid({
+    pageUid,
+    extensionAPI,
+  });
+  const commentHeaderEl = document.querySelector(
+    `.rm-block__input[id$="${commentsContainerUid}"]`
+  );
+  if (commentHeaderEl && commentsContainerUid) {
+    if (commentHeaderEl.hasAttribute("github-sync-loaded")) return;
+    commentHeaderEl.setAttribute("github-sync-loaded", "true");
+    const containerDiv = document.createElement("div");
+    containerDiv.className = "inline-block ml-2";
+    commentHeaderEl.appendChild(containerDiv);
+    ReactDOM.render(
+      <GitHubSyncPage
+        commentsContainerUid={commentsContainerUid}
+        extensionAPI={extensionAPI}
+      />,
+      containerDiv
+    );
+  }
+  commentHeaderEl;
+  handleTitleAdditions(h1, <TitleButtons pageUid={pageUid} />);
+};
+const formatComment = (c: GitHubComment) => {
+  const roamCreatedDate = window.roamAlphaAPI.util.dateToPageTitle(
+    new Date(c.created_at)
+  );
+  const commentHeader = `${c.user.login} on [[${roamCreatedDate}]]`;
+  const commentBody = c.body.trim();
+  return {
+    header: commentHeader,
+    body: commentBody,
+    props: {
+      "github-sync": {
+        comment: c,
+      },
+    },
+  };
+};
+
+// Components
+const CommentsComponent = ({ blockUid }: { blockUid: string }) => {
+  const [loading, setLoading] = useState(false);
+  const url = useMemo(() => {
+    const props = getBlockProps(blockUid) as GitHubCommentBlock;
+    const commentProps = props?.["github-sync"]?.["comment"];
+    return commentProps?.html_url;
+  }, [blockUid, loading]);
+
+  return (
+    <>
+      <Button
+        title={
+          !!url
+            ? "Comment was already sent. Click to view on GitHub."
+            : "Add to GitHub"
+        }
+        text={!!url ? "" : "Add to GitHub"}
```

Comment:
I'm using the text / no text based on their importance / how the user will use them.

It's very unlikely the user will want/need to click on settings before sending the page to GitHub.  Adding that text feels like it will just distract from the main action and just take up space.

On the other hand, removing the text would leave just out of context icon(s) by their selves under the title.